src/lib/pubkey/curve448/curve448_utils/curve448_scalar.cpp

```diff
@@ -0,0 +1,220 @@
+/*
+ * Ed448 Scalar
+ * (C) 2024 Jack Lloyd
+ *     2024 Fabian Albert - Rohde & Schwarz Cybersecurity
+ *
+ * Botan is released under the Simplified BSD License (see license.txt)
+ */
+#include <botan/internal/curve448_scalar.h>
+
+#include <botan/internal/mp_core.h>
+
+namespace Botan {
+
+namespace {
+constexpr size_t WORDS_REDUCE_SZ = words_for_bits(114 * 8);
+constexpr size_t WORDS_C = words_for_bits(28 * 8);
+
+/// @return (q,r) so that x = q*2^446 + r, r < L
+template <size_t S>
+auto div_mod_2_446(std::span<const word, S> x) {
+   if constexpr(S < WORDS_446) {
+      std::array<word, WORDS_446> r = {0};
+      copy_mem(std::span(r).template first<S>(), x);
+      return std::make_pair(std::array<word, 1>({0}), r);
+   } else {
+      std::array<word, WORDS_446> r;
+      copy_mem(r, std::span(x).template first<WORDS_446>());
+      // Clear the two most significant bits
+      r[WORDS_446 - 1] &= ~(word(0b11) << (sizeof(word) * 8 - 2));
+
+      constexpr size_t word_shift = 446 / (sizeof(word) * 8);
+      constexpr size_t bit_shift = 446 % (sizeof(word) * 8);
+
+      std::array<word, S - WORDS_446 + 1> q;
+      bigint_shr2(q.data(), x.data(), x.size(), word_shift, bit_shift);
+
+      return std::make_pair(q, r);
+   }
+}
+
+template <size_t S>
+consteval std::array<word, S / sizeof(word)> load_words(const std::array<uint8_t, S>& bytes)
+   requires(S % sizeof(word) == 0)
+{
+   // Currently load_le does not work with constexpr. Therefore, we have to use this workaround.
```

Comment:
This should work now: https://github.com/randombit/botan/pull/3869/commits/419424060650d07a151b3a275905e398fc9fe9a8

It didn't work, because the load/store functions use `std::memcpy` when no actual byte swaps are necessary. This does not work at compile-time, and probably won't any time soon. Also `std::bit_cast` (introduced in C++20) is not suitable for what we need (we want to transform a range of integers into another range of other integer types, for instance).

However, the load/store functions already come with a C++-only fallback (for platforms on which we cannot determine the endianness). The fix now uses the fallback also for the `constexpr` execution.

Reply:
Amazing, thanks!